snuba/datasets/querylog_processor.py

```diff
@@ -0,0 +1,72 @@
+from typing import Any, Mapping, Optional, Sequence, Union
+import uuid
+
+from snuba.processor import MessageProcessor, ProcessedMessage, ProcessorAction
+
+
+class QuerylogProcessor(MessageProcessor):
+    def get_request(self, request: Mapping[str, Any]) -> str:
+        return str({k: v for k, v in sorted(request.items())})
+
+    def get_sample(self, sample: Union[int, float]) -> float:
+        """
+        Since sample can be both integer or float between 0 and 1, we just cast
+        int to float so we have a common data type.
+        """
+        if isinstance(sample, int):
+            return float(sample)
+
+        return sample
+
+    def extract_query_list(self, query_list: Sequence[Mapping[str, Any]]) -> Mapping[str, Any]:
+        sql = []
+        status = []
+        final = []
+        cache_hit = []
+        sample = []
+        max_threads = []
+        duration_ms = []
+        trace_id = []
+
+        for query in query_list:
+            sql.append(query["sql"])
+            status.append(query["status"])
+            final.append(query["stats"].get("final", 0))
+            cache_hit.append(query["stats"].get("cache_hit", 0))
+            sample.append(query["stats"].get("sample", 0))
+            max_threads.append(query["stats"].get("max_threads", 0))
+            trace_id.append(str(uuid.UUID(query["trace_id"])))
+            # TODO: Calculate subquery duration, for now just insert 0s
+            duration_ms.append(0)
+
+        return {
```

Comment:
Please ensure with @JTCunning  that the amount of rows inserted per second (from our metrics) will be ok or whether we need to sample.

Reply:
Confirmed that we don't want to sample here